### DISCUSSION TITLE
Are you using MoltenVK for your Mac games?

### DISCUSSION THREAD

**Discussion by mrmacright** [KhronosGroup/MoltenVK#1402] [Q&A]
Hi there,

I own MrMacRight on YouTube, where I promote gaming on Mac. I'm doing a survey to see how many of you are using MoltenVK to port your games to Mac. 

Thanks!

**Comment by Rastafabisch** [comment]
I do, though this is done only indirectly using Wineskin (Wine wrapper). While now current moltenVK libraries are included as well as an dxvk installation tool I started experimenting with this quiet some back. However even than I wouldn’t have come that far without the work of [@gcenx](https://github.com/Gcenx/WineskinServer).

**Comment by billhollings** [comment]
General call for information:

What's the status for porting games from DX 9-12 through DXVK/vkd3d/Proton/Wine and MoltenVK to Apple platforms? 

Have any games made the full port yet through these routes (possibly with reduced capabilities), or are there outstanding issues that stop this from being possible?

**Reply by Gcenx** [reply]
> What's the status for porting games from DX 9-12 through DXVK/vkd3d/Proton/Wine and MoltenVK to Apple platforms?

DirectX9 still translates to OpenGL, DXVK-CX can’t be used due to the lack of required Vulkan features.

DirectX10/11 64Bit can use DXVK-CX or WineD3D Vulkan (used vkd3d shader compiler), however the lack of native transform-feedback and geometry shaders.

VKD3D - DirectX12 lightly isn’t even possible.

Upstream wine _currently_ doesn’t support 32Bit on 64Bit only versions of macOS also wine(32on64) doesn’t currently support Vulkan trunking anyway.

No Proton on macOS, Valve would need to enable that within the macOS Steam client but I don’t see that happening considering it seems Valve doesn’t really care about macOS anymore.

> Have any games made the full port yet through these routes (possibly with reduced capabilities), or are there outstanding issues that stop this from being possible?

CodeWeavers do a commercial port for Square for FINAL FANTASY XIV. That’s being using MoltenVK & DXVK-CX since 2019.

So depending on what a title requires it can actually run great, but something like the Witcher 3 will end up with invisible enemy’s (lack of transform feedback?) and graphical glitches (geometry shader?)

**Comment by rcaridade145** [comment]
> zink. Fixing vkGetPhysicalDeviceProperties2 and vkGetPhysicalDeviceFeatures2 for Vk 1.1 and VK_KHR_get_physical_device_properties2.
> MoltenVK does not export the vkGetPhysical2() functionns, even in Vulkan 1.2.154.0 

On Zink https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/7960

https://github.com/hrydgard/ppsspp/issues/13462

There is a work on progress also for the Yuzu emulator https://github.com/yuzu-emu/yuzu/pull/4165 .

**Reply by rcaridade145** [reply]
RPCS3 a PS3 emulator is adding MacOS support using MoltenVK - https://github.com/RPCS3/rpcs3/pull/11222 - to support the graphics emulation
